nebula_ros/src/hesai/hw_monitor_wrapper.cpp

```diff
@@ -217,71 +222,53 @@ void HesaiHwMonitorWrapper::OnHesaiLidarMonitorTimer()
   RCLCPP_DEBUG_STREAM(logger_, "OnHesaiLidarMonitorTimer END");
 }
 
-void HesaiHwMonitorWrapper::HesaiCheckStatus(
-  diagnostic_updater::DiagnosticStatusWrapper & diagnostics)
+void HesaiHwMonitorWrapper::HesaiCheckStatus(diagnostic_updater::DiagnosticStatusWrapper & diagnostics)
 {
   std::scoped_lock lock(mtx_lidar_status_);
   if (current_status_) {
-    uint8_t level = diagnostic_msgs::msg::DiagnosticStatus::OK;
-    std::vector<std::string> msg;
-
-    diagnostics.add("system_uptime", std::to_string(current_status_->system_uptime.value()));
-    diagnostics.add("startup_times", std::to_string(current_status_->startup_times.value()));
-    diagnostics.add(
-      "total_operation_time", std::to_string(current_status_->total_operation_time.value()));
-
-    diagnostics.summary(level, boost::algorithm::join(msg, ", "));
+    json data = current_status_->to_json();
+    for (auto & [key, value] : data.items()) {
```

Comment:
Please make this `const auto &` since it is not modified. This makes it clearer to the reader and the compiler, that nothing is modified in this loop. Please also add it to the other loops in this file.
```suggestion
    for (const auto & [key, value] : data.items()) {
```